mongoengine/queryset/transform.py

```diff
@@ -47,6 +47,9 @@ def query(_doc_cls=None, _field_operation=False, **query):
         if len(parts) > 1 and parts[-1] in MATCH_OPERATORS:
             op = parts.pop()
 
+        if len(parts) > 1 and not parts[-1]:
```

Comment:
Would you consider making it explicit? something like:
`and parts[-1] == ""`
In addition, an inline comment of why this condition is here could be useful